src/components/tx/CreateTx/create.js

```diff
@@ -126,47 +103,90 @@ const CreateTx = connect(
+                }
+
+                const decimals = convert.toNumber(tokenInfo.get('decimals'));
+                const tokenUnits: BigNumber = convert.toBaseUnits(convert.toBigNumber(data.value), decimals);
+                const txData = Tokens.actions.createTokenTxData(
+                    data.to,
+                    tokenUnits,
+                    data.isTransfer);
+
+                tx = {
+                    from: data.from,
+                    gasPrice: toHex(data.gasPrice.value()),
+                    gas: toHex(data.gas),
+                    to: data.token,
+                    value: convert.toHex(0),
+                    data: txData,
+                };
+            } else {
+                // Ordinary Tx
+                tx = {
+                    from: data.from,
+                    gasPrice: toHex(data.gasPrice.value()),
+                    gas: toHex(data.gas),
+                    to: data.to,
+                    value: toHex(etherToWei(data.value)),
+                    data: '',
+                };
+            }
+
+            log.debug(JSON.stringify(tx));
+
+
+            // 2. Validate Trace and then Send TX
+            return traceValidate(tx, dispatch)
+                .then((estimatedGas) => {
+                    log.debug(`Tx validated by trace. Estimated Gas ${estimatedGas}`);
+
                     dispatch(setWatch(false));
-                    closeConnection().then(() => {
+                    return closeConnection().then(() => {
                         if (useLedger) {
-                            dispatch(screen.actions.showDialog('sign-transaction', data));
+                            return dispatch(screen.actions.showDialog('sign-transaction', data));
                         }
-                        dispatch(
+                        return dispatch(
                             accounts.actions.sendTransaction(
-                                data.from,
+                                tx.from,
                                 data.password,
-                                data.to,
-                                toHex(data.gas),
-                                toHex(data.gasPrice.value()),
-                                toHex(etherToWei(data.value)))
+                                tx.to,
+                                tx.gas,
+                                tx.gasPrice,
+                                tx.value,
+                                tx.data)
                         );
                     });
                 })
                 .catch((err) => {
                     log.error('Validation failure', err);
-                    throw new SubmissionError(err);
+                    throw new SubmissionError({ _error: (err.message || JSON.stringify(err)) });
                 });
         },
         onChangeAccount: (all, value) => {
             // load account information for selected account
```

Comment:
why removed?

Reply:
This is a feature of redux-form. If you provide object with `_error` property it will populate `props.error` for react component.